vms/administrator/tests/test_unit.py

```diff
@@ -73,21 +73,92 @@ def test_valid_model_create(self):
         self.assertEqual(admin_in_db.state, admin[7])
         self.assertEqual(admin_in_db.country, admin[8])
         self.assertEqual(admin_in_db.phone_number, admin[9])
-        self.assertEqual(admin_in_db.unlisted_organization, admin[10])
+        self.assertEqual(str(admin_in_db.organization), admin[10])
 
     def test_invalid_model_create(self):
         """
         Database test for model creation with invalid values.
         """
+        # Admin first name invalid
         admin = [
-            'admin-username', 'admin-password!@#$%^&*()_', 'admin!first!name',
-            'admin-last-name', 'admin-email1@systers.org', 'admin-address',
-            'admin-city', 'admin-state', 'admin-country', '9999999999',
-            'admin-org'
+            'adminusername1', 'admin-password!@#$%^&*()_', 'admin~first~name',
+            'admin last name', 'adminemail2@systers.org', 'admin address',
+            'admin city', 'admin state', 'admin country', '9999999999',
+            'admin org2'
+        ]
+        created_admin = create_admin_with_details(admin)
+        self.assertRaisesRegexp(ValidationError, BasePage.ENTER_VALID_VALUE, created_admin.full_clean)
+
+        # Admin last name invalid
```

Comment:
Instead of creating one test which checks all the invalid cases, can we have multiple tests each of which is responsible for one single check ?

Reply:
Okay will do it. :+1: in the other PR as well. #812

Reply:
Great !!

Reply:
Done in this PR. please review again @MehaKaushik